src/preview/VideoPreview.test.tsx

```diff
@@ -0,0 +1,190 @@
+	});
+
+	it('should not render the video when canPlayType return empty string on mime type(mime type not supported)', () => {
+		jest.spyOn(utils, 'videoCanPlayType').mockReturnValue('');
+
+		setup(
+			<VideoPreview
+				onClose={jest.fn()}
+				show
+				src={''}
+				filename="video name"
+				mimeType={'unsupported/mimeType'}
+			/>
+		);
+
+		expect(screen.getByText(cannotBePlayedMessage)).toBeVisible();
+		expect(screen.queryByTestId('video')).not.toBeInTheDocument();
+	});
+
+	it('should render the video when mime type props is not provided', () => {
+		setup(<VideoPreview onClose={jest.fn()} show src={''} filename="video name" />);
+
+		expect(screen.getByTestId('video')).toBeVisible();
+		expect(screen.queryByText(cannotBePlayedMessage)).not.toBeInTheDocument();
+	});
+
+	it('should render the video when canPlayType return maybe string on mime type', () => {
+		jest.spyOn(utils, 'videoCanPlayType').mockReturnValue('maybe');
+		const mimeType = 'video/mp4';
+		setup(<VideoPreview onClose={jest.fn()} show src={''} mimeType={mimeType} />);
+		expect(screen.getByTestId('video')).toBeVisible();
+		expect(screen.queryByText(cannotBePlayedMessage)).not.toBeInTheDocument();
+	});
+
+	it('should render the video when canPlayType return probably string on mime type', () => {
+		jest.spyOn(utils, 'videoCanPlayType').mockReturnValue('probably');
+		const mimeType = 'video/mp4';
+		setup(<VideoPreview onClose={jest.fn()} show src={''} mimeType={mimeType} />);
+		expect(screen.getByTestId('video')).toBeVisible();
+		expect(screen.queryByText(cannotBePlayedMessage)).not.toBeInTheDocument();
+	});
+
+	it('should render the fail string when video request fails', async () => {
+		setup(<VideoPreview onClose={jest.fn()} show src={''} />);
+		fireEvent.error(screen.getByTestId('video'));
+		expect(await screen.findByText(cannotBePlayedMessage)).toBeVisible();
+	});
+
+	it('should call video play when keyboard space is clicked', async () => {
+		jest.spyOn(HTMLMediaElement.prototype, 'paused', 'get').mockReturnValue(true);
+		const playStub = jest
+			.spyOn(window.HTMLVideoElement.prototype, 'play')
+			.mockImplementation(() => Promise.resolve());
+
+		const pauseStub = jest
+			.spyOn(window.HTMLVideoElement.prototype, 'pause')
+			.mockImplementation(() => undefined);
+
+		const { user } = setup(<VideoPreview onClose={jest.fn()} show src={''} />);
+		await user.keyboard(' ');
+		expect(playStub).toHaveBeenCalled();
+		expect(pauseStub).not.toHaveBeenCalled();
+		jest.restoreAllMocks();
+	});
+
+	it('should call video pause when video is not paused and keyboard space is clicked', async () => {
+		jest.spyOn(HTMLMediaElement.prototype, 'paused', 'get').mockReturnValue(false);
+		const pauseStub = jest
+			.spyOn(window.HTMLVideoElement.prototype, 'pause')
+			.mockImplementation(() => undefined);
+
+		const playStub = jest
+			.spyOn(window.HTMLVideoElement.prototype, 'play')
+			.mockImplementation(() => Promise.resolve());
```

Comment:
The defined mock is not useful
```suggestion
			.mockImplementation();
```